test/NuGet.Core.Tests/NuGet.Packaging.Test/DependenciesGroupsForEachTFMRuleTests.cs

```diff
@@ -0,0 +1,446 @@
+    {
+        [Fact]
+        public void GenerateWarnings_PackageWithDependenciesForEachTFMInLib_ShouldNotWarn()
+        {
+            //Arrange
+            var frameworks = new NuGetFramework[]
+            {
+                FrameworkConstants.CommonFrameworks.Net2,
+                FrameworkConstants.CommonFrameworks.Net35,
+                FrameworkConstants.CommonFrameworks.Net4,
+                FrameworkConstants.CommonFrameworks.Net45,
+                FrameworkConstants.CommonFrameworks.NetStandard10,
+                FrameworkConstants.CommonFrameworks.NetStandard13,
+                FrameworkConstants.CommonFrameworks.NetStandard20,
+            };
+
+            var fileStrings = new string[]
+            {
+                "lib/net20/test.dll",
+                "lib/net35/test.dll",
+                "lib/net40/test.dll",
+                "lib/net45/test.dll",
+                "lib/netstandard1.0/test.dll",
+                "lib/netstandard1.3/test.dll",
+                "lib/netstandard2.0/test.dll",
+            };
+
+            // Act
+            var rule = new DependenciesGroupsForEachTFMRule();
+            var (compat, file, nuspec) = rule.Categorize(fileStrings, frameworks);
+            var issues = rule.GenerateWarnings(compat, file, nuspec).ToList();
+
+            // Assert
+            Assert.False(issues.Any(p => p.Code == NuGetLogCode.NU5128));
+            Assert.False(issues.Any(p => p.Code == NuGetLogCode.NU5130));
+        }
+
+        [Fact]
+        public void GenerateWarnings_PackageWithDependenciesInRefHasExactMatch_ShouldNotWarn()
+        {
+            var frameworks = new NuGetFramework[]
+            {
+                FrameworkConstants.CommonFrameworks.Net2,
+                FrameworkConstants.CommonFrameworks.Net35,
+                FrameworkConstants.CommonFrameworks.Net4,
+                FrameworkConstants.CommonFrameworks.Net45,
+                FrameworkConstants.CommonFrameworks.NetStandard10,
+                FrameworkConstants.CommonFrameworks.NetStandard13,
+                FrameworkConstants.CommonFrameworks.NetStandard20,
+            };
+
+            //Arrange
+            var fileStrings = new string[]
+            {
+                "ref/net20/test.dll",
+                "ref/net35/test.dll",
+                "ref/net40/test.dll",
+                "ref/net45/test.dll",
+                "ref/netstandard1.0/test.dll",
+                "ref/netstandard1.3/test.dll",
+                "ref/netstandard2.0/test.dll",
+            };
+
+            // Act
+            var rule = new DependenciesGroupsForEachTFMRule();
+            var (compat, file, nuspec) = rule.Categorize(fileStrings, frameworks);
+            var issues = rule.GenerateWarnings(compat, file, nuspec).ToList();
+
+            // Assert
+            Assert.False(issues.Any(p => p.Code == NuGetLogCode.NU5128));
+            Assert.False(issues.Any(p => p.Code == NuGetLogCode.NU5130));
```

Comment:
Also FYI, if you don't have time, I think it's better to be more precise about what you expect. `Assert.Equal(0, issues.Count)`. Otherwise this test is saying it's ok to generate a warning, as long as it's neither `NU5128` or `NU5130`.